queues/queue.go

```diff
@@ -0,0 +1,34 @@
+package queues
+
+import "github.com/akitasoftware/go-utils/optionals"
+
+type Queue[T any] struct {
+	elements []T
+}
+
+func NewQueue[T any](elements ...T) *Queue[T] {
+	return &Queue[T]{
+		elements: elements,
+	}
+}
+
+func (q *Queue[T]) Enqueue(t T) {
+	q.elements = append(q.elements, t)
+}
+
+func (q *Queue[T]) Dequeue() optionals.Optional[T] {
+	if len(q.elements) == 0 {
+		return optionals.None[T]()
+	}
+	result := q.elements[0]
+	q.elements = q.elements[1:]
```

Comment:
I am deeply suspicious about the practical performance of these as implemented by Go.

There are a couple discussions online around the fact that the memory for [0] will not be freed until enough appends() have been done to cause a reallocation.  If we're to have a general-purpose implementation I would prefer that we have a ring-buffer based approach.

And, of course, there's always channels.  :)

Reply:
Added a comment to warn about memory performance in 446e7a6. Channels can work, but once created, their capacity can't be changed.